libCacheSim/dataStructure/hashtable/chainedHashTableV2.c

```diff
@@ -248,26 +248,37 @@ bool chained_hashtable_try_delete_v2(hashtable_t *hashtable,
   return false;
 }
 
-void chained_hashtable_delete_obj_id_v2(hashtable_t *hashtable,
+bool chained_hashtable_delete_obj_id_v2(hashtable_t *hashtable,
                                         const obj_id_t obj_id) {
-  hashtable->n_obj -= 1;
-  uint64_t hv = get_hash_value_int_64(obj_id) & hashmask(hashtable->hashpower);
-  cache_obj_t *cache_obj = hashtable->ptr_table[hv];
-  if (cache_obj != NULL && cache_obj->obj_id == obj_id) {
-    hashtable->ptr_table[hv] = cache_obj->hash_next;
-    if (!hashtable->external_obj) free_cache_obj(cache_obj);
-    return;
-  }
 
-  cache_obj = cache_obj->hash_next;
-  while (cache_obj != NULL && cache_obj->obj_id != obj_id) {
-    cache_obj = cache_obj->hash_next;
+  uint64_t hv = get_hash_value_int_64(&obj_id) & hashmask(hashtable->hashpower);
+  cache_obj_t *cur_obj = hashtable->ptr_table[hv];
+  // the hash bucket is empty
+  if(cur_obj == NULL) return false;
+
+  // the object to remove is the first object in the hash bucket
+  if (cur_obj->obj_id == obj_id) {
+    hashtable->ptr_table[hv] = cur_obj->hash_next;
+    if (!hashtable->external_obj) free_cache_obj(cur_obj);
+    hashtable->n_obj -= 1;
+    return true;
   }
 
-  if (cache_obj != NULL) {
-    cache_obj->hash_next = cache_obj->hash_next;
-    if (!hashtable->external_obj) free_cache_obj(cache_obj);
+  cache_obj_t *prev_obj;
+
+  do {
+    prev_obj = cur_obj;
+    cur_obj = cur_obj->hash_next;
+  } while(cur_obj != NULL && cur_obj->obj_id != obj_id);
+
+  if (cur_obj != NULL) {
+    prev_obj->hash_next = cur_obj->hash_next;
+    if (!hashtable->external_obj) free_cache_obj(cur_obj);
+    hashtable->n_obj -= 1;
```

Comment:
It looks like there was a bug in the old code. Good catch! Can you port this change to the develop branch?

Reply:
Of course I will. This is just a minor bug, because it seems that this function is not be used in the project.

Reply:
Thank you!